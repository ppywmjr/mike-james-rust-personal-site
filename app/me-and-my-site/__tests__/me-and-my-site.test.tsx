import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import MeAndMySite from "../page";

describe("Me and my site", () => {
  it("renders with h1", () => {
    render(<MeAndMySite />);

    const nav = screen.getByRole("heading", { level: 1 });

    expect(nav).toBeVisible();
  });

  it("renders with img", () => {
    render(<MeAndMySite />);

    const image = screen.getByRole("img");

    expect(image).toBeVisible();
  });

  it("renders with text", () => {
    render(<MeAndMySite />);

    const text = screen.getByText(/Hi folks/);

    expect(text).toBeVisible();
  });
});
