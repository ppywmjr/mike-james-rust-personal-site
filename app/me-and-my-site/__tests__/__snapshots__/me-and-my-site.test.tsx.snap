// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me and my site renders the page 1`] = `
<div>
  <h1
    className="text-3xl font-bold my-6 md:mt-8 md:mb-12"
  >
    Me and my site
  </h1>
  <div
    className="flex flex-row flex-wrap"
  >
    <div
      className="basis-24 md:basis-32 shrink-0 mr-6 mb-6"
    >
      <img
        alt="My selfie. A handsome chap with long wavy hair, a tidy beard and a big smile"
        className="rounded-lg"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height={143}
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fprofile.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fprofile.jpg&w=128&q=75 1x, /_next/image?url=%2Fprofile.jpg&w=256&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={120}
      />
    </div>
    <p
      className="basis-40 shrink-0 mb-6 text-xl italic"
    >
      An intro to what this site is all about
    </p>
  </div>
  <p
    className="max-w-prose leading-loose"
  >
    Hi folks, I'm Mike and this is my site. Welcome! This site is primarily for me, but I'm delighted to imagine you reading it. I've made a few little apps just for fun over the years but I'm motivated to make this one for a couple of new reasons. Firstly, I know someone who wants to build a website and they came to me for some advice. I found that as much as I know how to make a huge difference to agile teams, I realised I don't really know how to put together a complete website. So I was a little embarrased. Yes, my core motivation is embarrassment. The second reason is the technology. I've been working as a tester with Nextjs apps for a couple of years now and it's so easy to work with that I figured the effort to get something up and running would be minimal. Also, Nextjs has so many features that I think it would be really and fun to try out some different approaches on different pages.
  </p>
</div>
`;
